src/jit/emitinl.h

```diff
@@ -388,6 +421,35 @@ ssize_t             emitter::emitGetInsAmdAny(instrDesc *id)
                              regmask |= RBM_R11;
 #endif
 
+#ifdef _TARGET_ARM64_
+    assert(REGNUM_BITS >= 5);
+    encodeMask = id->idReg1();
+
+    if (encodeMask & 0x01)
+        regmask |= RBM_R19;
+    if (encodeMask & 0x02)
+        regmask |= RBM_R20;
+    if (encodeMask & 0x04)
+        regmask |= RBM_R21;
+    if (encodeMask & 0x08)
+        regmask |= RBM_R22;
+    if (encodeMask & 0x10)
+        regmask |= RBM_R23;
+
+    encodeMask = id->idReg2();
+
+    if (encodeMask & 0x01)
+        regmask |= RBM_R24;
+    if (encodeMask & 0x02)
+        regmask |= RBM_R25;
+    if (encodeMask & 0x04)
+        regmask |= RBM_R26;
+    if (encodeMask & 0x08)
+        regmask |= RBM_R27;
+    if (encodeMask & 0x10)
+        regmask |= RBM_R28;
+#endif
```

Comment:
I agree the #ifdefs don't look awesome but I went looking in the code emit.h etc. and was confused as to what best practice was.  I see lots of #ifdef in the main header, sometimes inline inline implementation there, sometimes the code is in the target specific header,  sometimes in the inline header.  What's the best way for me to think about how to remove this #ifdef?